src/audio_unit/pipeline.rs

```diff
@@ -34,16 +32,8 @@ impl Pipeline {
     }
 }
 
-impl Display for Pipeline {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let mut names: Vec<_> = self
-            .audio_units
-            .iter()
-            .map(|audio_unit| audio_unit.name())
-            .collect();
-        names.insert(0, "Input".into());
-        names.insert(names.len(), "Output".into());
-
-        names.join(" -> ").fmt(f)
+impl AudioUnit for Pipeline {
+    fn process(&mut self, input: &[f32], output: &mut [f32]) -> Result<()> {
+        self.process(input, output)
```

Comment:
Ok so this is saying "now pipelines can themselves be treated as audio units" and there basically already was a symmetrical interface (`process()`)? I'm confused about the name overloading of the two identically-signatured implementations of `process()`?

Reply:
> Ok so this is saying "now pipelines can themselves be treated as audio units" and there basically already was a symmetrical interface (process())?

Yes.

Re: name overloading: since everything is statically compiled, there's no function lookup via name happening at runtime, so it works to overload function names like this. (And there is some way of specifying which one you want at the call site, although I can't remember it off the top of my head.) It's probably not a good idea though, at least in cases like this where the functions are identical, and one just delegates its implementation to the other. We should just leave the `AudioUnit` version of it, I think. I've already done that in my WIP MIDI PR, so I'll leave it here for now, though. One downside of doing that is that you need to import `Pipeline` and the `AudioUnit` trait to use that function, but that seems fine in this case.